gnome-image-installer/pages/install/gis-install-page.c

```diff
@@ -421,8 +429,55 @@ gis_install_page_verify (GisPage *page)
 
   g_child_watch_add (priv->gpg, (GChildWatchFunc)gis_install_page_gpg_watch, page);
 
-  g_free (args[ARGS - 2]);
-  g_free (args[ARGS - 3]);
+out:
+  g_free (image_path);
+  g_free (signature_path);
+  g_object_unref (signature);
+}
+
+static void
+gis_install_page_gpg_import_complete (GObject      *source,
+                                      GAsyncResult *result,
+                                      gpointer      user_data)
+{
+  GSubprocess *gpg_import = G_SUBPROCESS (source);
+  GisPage *page = GIS_PAGE (user_data);
+
+  if (g_subprocess_wait_check_finish (gpg_import, result, NULL))
+    {
+      gis_install_page_verify (page);
+    }
+  else
+    {
+      gint code = g_subprocess_get_exit_status (gpg_import);
+      GError *error = g_error_new (GIS_INSTALL_ERROR, code,
+                                   _("Could not import Endless OS key"));
+      gis_install_page_verify_failed (page, error);
+    }
+
+  g_object_unref (gpg_import);
+}
+
+
+static gboolean
+gis_install_page_prepare_verify (GisPage *page)
+{
+
+  GSubprocess *gpg_import;
+  GError *error = NULL;
+
+  gpg_import = g_subprocess_new (G_SUBPROCESS_FLAGS_NONE, &error,
+      "gpg", "--import", IMAGE_KEYRING, NULL);
```

Comment:
Actually, shouldn't the signing key already be in the repo? If so, importing is redundant.